ExecuteJobs.java

```diff
@@ -0,0 +1,124 @@
+import java.util.stream.*;
+import java.util.*;
+import java.io.*;
+//import java.util.Stream;
+
+public class ExecuteJobs {
+  private static final int ERR_SUCCESS = 0;
+  private static final int ERR_FAILED_TO_KILL = 1;
+  private static final int ERR_FAILED_TO_FIND = 2;
+  private static final int ERR_FAILED_TO_START = 3;
+
+  public ErrorInfo start(String sProcess) throws IOException {
+    ErrorInfo errorInfo = new ErrorInfo();
+    Process process = Runtime.getRuntime().exec(sProcess);
+
+    ErrorInfo errorStatus = getProcessStatus(process.pid());
+    if(errorStatus.getIoMessage() == null) {
+      errorInfo.setErrorCode(ERR_FAILED_TO_START);
+      errorInfo.setIoMessage("Failed to start " + sProcess);
+      return errorInfo;
+    }
+    errorInfo.setErrorCode(ERR_SUCCESS);
+    errorInfo.setIoMessage("Successfully start " + sProcess);
+    errorInfo.setPid(process.pid());
+    return errorInfo;
+  }
+
+  public ErrorInfo stop(long pid) throws IOException {
+    ErrorInfo errorInfo = new ErrorInfo();
+
+    ErrorInfo errorStatus = getProcessStatus(pid);
+    if(errorStatus.getIoMessage() == null) {
+      errorInfo.setErrorCode(ERR_FAILED_TO_FIND);
+      errorInfo.setIoMessage("Failed to find PID " + pid + ". PID was not killed.");
+      return errorInfo;
+    }
+
+    Runtime.getRuntime().exec("kill " + pid);
+    errorStatus = getProcessStatus(pid);
+    if(errorStatus.getIoMessage() != null) {
+      errorInfo.setErrorCode(ERR_FAILED_TO_KILL);
+      errorInfo.setIoMessage("Failed to kill " + pid);
+      return errorInfo;
+    }
+
+    errorInfo.setErrorCode(ERR_SUCCESS);
+    errorInfo.setIoMessage("Successfully killed " + pid);
+    return errorInfo;     
+  }
+
+  public ErrorInfo query(long pid) throws IOException {
+    ErrorInfo errorInfo = new ErrorInfo();
+
+    ErrorInfo errorStatus = getProcessStatus(pid);
+    if(errorStatus.getIoMessage() == null) {
+      errorInfo.setErrorCode(ERR_FAILED_TO_FIND);
+      errorInfo.setIoMessage("Failed to find PID " + pid);
+      errorInfo.setErrorMessage(errorStatus.getIoMessage());
+      return errorInfo;
+    }
+    errorInfo.setErrorCode(ERR_SUCCESS);
+    errorInfo.setIoMessage("PID " + pid + " status: " + errorStatus.getIoMessage());
+    errorInfo.setErrorMessage(errorStatus.getIoMessage());
+    return errorInfo;
+  }
+
+  public ArrayList<ErrorInfo> currentJobStatus(ArrayList<ErrorInfo> pids) throws IOException {
+    ArrayList<ErrorInfo> statuses = new ArrayList<ErrorInfo>();
+
+    int i = 0;
+    while(i < pids.size()) {
+      Long pid = pids.get(i).getPid();
```

Comment:
Looks like this method is only concerned with a list of pids - why does it receive a list of `ErrorInfo` as parameter?

Reply:
Each ErrorInfo object contains the pids needed. An up-to-date ArrayList<ErrorInfo> is maintained in the Server class. Initially I only stored Longs in the list, but after modifying start() to return an ErrorInfo object (which is necessary for building the Response object), I decided that it made more sense to store the entire ErrorInfo object in the ArrayList instead of parsing the returned ErrorInfo object to get the pid.